### PR TITLE
Fixes Contract deployment and permission granting

### PR DESCRIPTION
**Pull request by mathewmeconry**
## Description

Bundles the permission granting during deployment into 1 file.
Fixes goerli network configuration
removes subgraph not null enforcing on Proposal.metadata

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

**Review by Rekard0 — APPROVED** [review]
LGTM

File: packages/contracts/deploy/60_dao-factory.ts [hunk 1]
```diff
@@ -1,39 +1,22 @@
...
+import { getContractAddress } from './helpers';
 
 const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
-  const {deployments, getNamedAccounts, ethers} = hre;
-  const {deploy} = deployments;
+  const { deployments, getNamedAccounts, ethers } = hre;
+  const { deploy } = deployments;
 
-  const {deployer} = await getNamedAccounts();
+  const { deployer } = await getNamedAccounts();
```
Comment (Michael-A-Heuer) [inline packages/contracts/deploy/60_dao-factory.ts:10]: Your IDE doesn't seem to use our `.prettierrc` in which we specify `"bracketSpacing": false`.
We use this plugin for VScode: https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

File: packages/contracts/CHANGELOG.md [hunk 2]
```diff
@@ -25,6 +25,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added deployment script for `managing-dao`, `plugin-registry` and `pluginRepo-factory`.
 - Added an abstract `ERC165RegistryBase` `Component` to register contracts by their address based on their ERC165 interface ID.
 - Added a concrete `ERC165Registry` implementation.
+- Added goerli configuration to deploy to goerli
```
Comment (Michael-A-Heuer) [inline packages/contracts/CHANGELOG.md:28]:
```suggestion
- Added goerli configuration to deploy to goerli.
```

File: packages/contracts/deploy/70_permissions.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    const managingDAOAddress = await getContractAddress('DAO', hre);
+    const managingDaoContract = await ethers.getContractAt(
+        'DAO',
+        managingDAOAddress
+    );
+    const daoRegistryAddress = await getContractAddress('DAORegistry', hre);
+
+    console.log('Granting permissions')
+    // Grant REGISTER_PERMISSION_ID to pluginRepo factory
```
Comment (Michael-A-Heuer) [inline packages/contracts/deploy/70_permissions.ts:21]:
```suggestion
```

File: packages/contracts/deploy/70_permissions.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    const REGISTER_PERMISSION_ID = ethers.utils.id('REGISTER_PERMISSION');
+    const aragonPluginRegistryAddress = await getContractAddress('AragonPluginRegistry', hre);
+    const pluginRepoFactoryAddress = await getContractAddress('PluginRepoFactory', hre);
+
+    console.log(`Granting REGISTER_PERMISSION_ID for pluginRepo factory (${pluginRepoFactoryAddress})`)
```
Comment (Michael-A-Heuer) [inline packages/contracts/deploy/70_permissions.ts:26]:
```suggestion
    console.log('Granting the `REGISTER_PERMISSION_ID` permission to the `PluginRepoFactory` (${pluginRepoFactoryAddress})').
```

File: packages/contracts/deploy/70_permissions.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        pluginRepoFactoryAddress,
+        REGISTER_PERMISSION_ID
+    );
+    await grantTx.wait();
+
+    // Grant REGISTER_ENS_SUBDOMAIN_PERMISSION to daoRegistry
+    const REGISTER_ENS_SUBDOMAIN_PERMISSION_ID = ethers.utils.id("REGISTER_ENS_SUBDOMAIN_PERMISSION")
+    const ensSubdomainRegistrarAddress = await getContractAddress('ENSSubdomainRegistrar', hre);
+
+    console.log(`Granting REGISTER_ENS_SUBDOMAIN_PERMISSION for daoRegistry (${daoRegistryAddress})`)
```
Comment (Michael-A-Heuer) [inline packages/contracts/deploy/70_permissions.ts:38]:
```suggestion
    console.log(`Granting the `REGISTER_ENS_SUBDOMAIN_PERMISSION_ID` permission to  the `DaoRegistry` (${daoRegistryAddress})`)
```

File: packages/contracts/deploy/70_permissions.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        daoRegistryAddress,
+        REGISTER_ENS_SUBDOMAIN_PERMISSION_ID
+    );
+    await grantTx.wait();
+
+    // Grant REGISTER_DAO_PERMISSION_ID to daoRegistry
+    const REGISTER_DAO_PERMISSION_ID = ethers.utils.id('REGISTER_DAO_PERMISSION');
+    const daoFactoryAddress = await getContractAddress('DAOFactory', hre);
+
+    console.log(`Granting REGISTER_DAO_PERMISSION for daoRegistry (${daoRegistryAddress})`)
```
Comment (Michael-A-Heuer) [inline packages/contracts/deploy/70_permissions.ts:50]:
```suggestion
    console.log(`Granting the REGISTER_DAO_PERMISSION_ID permission to daoRegistry (${daoRegistryAddress})`)
```

File: packages/subgraph/CHANGELOG.md [hunk 7]
```diff
@@ -23,6 +23,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Refactored event names.
 - Refacotred `deposits` and `withdraws` into one field `transfers`
 - Refactored `VaultWithdraw` and `VaultDeposit` into one type `VaultTransfer`
+- Removes not null enforcing `Proposal.metadata`
```
Comment (Michael-A-Heuer) [inline packages/subgraph/CHANGELOG.md:26]:
```suggestion
- Refactored `deposits` and `withdraws` into one field `transfers`
- Refactored `VaultWithdraw` and `VaultDeposit` into one type `VaultTransfer`.
- Removes not null enforcing `Proposal.metadata`.
```

**Review by Michael-A-Heuer — COMMENTED** [review]
Looks good. Minor suggestions related to style and formatting.

**Review by Michael-A-Heuer — APPROVED** [review]

**Review by Rekard0 — APPROVED** [review]

**Pull request merged**